entrypoints/popup/App.tsx

```diff
@@ -1,6 +1,11 @@
 import { RouterProvider } from "react-router-dom";
 import { router } from "@/entrypoints/popup/router.tsx";
+import { LedgerTransportProvider } from "@/contexts/LedgerTransportContext.tsx";
 
 export default function App() {
-  return <RouterProvider router={router} />;
+  return (
+    <LedgerTransportProvider>
```

Comment:
The Ledger context must be placed outside the router; otherwise, the transport will be lost when switching pages.

I also suggest keeping PriceProvider, RpcClientProvider, and similar providers outside the router to prevent them from reloading when navigating between pages.

Reply:
Interesting, please add a comment in the file (I don't like comments, but when is a tricky thing to understand, let's just save some time to our future selves).

Reply:
Also I ran into this issues before with `react-router`. Most of the time it will only do a shallow redirection (not re-rendering the all hierarchy of components). But the exception happens when we use search params in the URL.

Here is what I found last time https://github.com/remix-run/react-router/discussions/9851.

The devs do not agree with each others.

I will note this issues and test everything before moving every provider here.

Reply:
https://github.com/forbole/kastle/pull/41/commits/71b44b9871209d90d9449b13174d6cb03f0db7bf